BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java

```diff
@@ -131,28 +132,40 @@ public void receiveDetections(Detector.Detections<Barcode> detections) {
   }
 
   private void handleIsbnInput(String isbn) {
-    isbnRetriever = new IsbnRetriever(isbn);
-    System.out.println(isbnRetriever);
-    thread = new Thread(isbnRetriever);
-    thread.start();
-
-    try {
-      thread.join();
-    } catch (Exception e) {
-      System.out.println(e);
-    }
+    String cleanIsbn = isbn.replaceAll("\\s", "");
+
+    if (dataValidation.isValidIsbn10or13(cleanIsbn)) {
+      isbnRetriever = new IsbnRetriever(cleanIsbn);
+      System.out.println(isbnRetriever);
+      thread = new Thread(isbnRetriever);
+      thread.start();
+
+      try {
+        thread.join();
+      } catch (Exception e) {
+        System.out.println(e);
```

Comment:
```suggestion
        e.printStackTrace();
```
Replace the custom system-print with a printstacktrace for logging exceptions.

Reply:
Done.